src/biz/collab/utils.rs

```diff
@@ -0,0 +1,278 @@
+  for field in fields {
+    let field_id: String = field.id.clone();
+    let type_option_reader: Box<dyn TypeOptionCellWriter> = {
+      let field_type: &FieldType = &FieldType::from(field.field_type);
+      let type_option_data: TypeOptionData = match field.type_options.get(&field_type.type_id()) {
+        Some(tod) => tod.clone(),
+        None => HashMap::new(),
+      };
+      type_option_cell_writer(type_option_data, field_type)
+    };
+    type_option_reader_by_id.insert(field_id, type_option_reader);
+  }
+  type_option_reader_by_id
+}
+
+/// create a map type option reader by field id
+pub fn type_option_reader_by_id(
+  fields: &[Field],
+) -> HashMap<String, Box<dyn TypeOptionCellReader>> {
+  let mut type_option_reader_by_id: HashMap<String, Box<dyn TypeOptionCellReader>> =
+    HashMap::with_capacity(fields.len());
+  for field in fields {
+    let field_id: String = field.id.clone();
+    let type_option_reader: Box<dyn TypeOptionCellReader> = {
+      let field_type: &FieldType = &FieldType::from(field.field_type);
+      let type_option_data: TypeOptionData = match field.type_options.get(&field_type.type_id()) {
+        Some(tod) => tod.clone(),
+        None => HashMap::new(),
+      };
+      type_option_cell_reader(type_option_data, field_type)
+    };
+    type_option_reader_by_id.insert(field_id, type_option_reader);
+  }
+  type_option_reader_by_id
+}
+
+pub fn type_options_serde(
+  type_options: &TypeOptions,
+  field_type: &FieldType,
+) -> HashMap<String, serde_json::Value> {
+  let type_option = match type_options.get(&field_type.type_id()) {
+    Some(type_option) => type_option,
+    None => return HashMap::new(),
+  };
+
+  let mut result = HashMap::with_capacity(type_option.len());
+  for (key, value) in type_option {
+    match field_type {
+      FieldType::SingleSelect | FieldType::MultiSelect | FieldType::Media => {
+        if let yrs::Any::String(arc_str) = value {
+          if let Ok(serde_value) = serde_json::from_str::<serde_json::Value>(arc_str) {
+            result.insert(key.clone(), serde_value);
+          }
+        }
+      },
+      _ => {
+        result.insert(key.clone(), serde_json::to_value(value).unwrap_or_default());
+      },
+    }
+  }
+
+  result
+}
+
+pub async fn get_database_body(
+  collab_storage: &CollabAccessControlStorage,
+  workspace_uuid_str: &str,
+  database_uuid_str: &str,
+) -> Result<(Collab, DatabaseBody), AppError> {
+  let db_collab = get_latest_collab(
+    collab_storage,
+    GetCollabOrigin::Server,
+    workspace_uuid_str,
+    database_uuid_str,
+    CollabType::Database,
+  )
+  .await?;
+  let db_body = DatabaseBody::from_collab(
+    &db_collab,
+    Arc::new(NoPersistenceDatabaseCollabService),
+    None,
+  )
+  .ok_or_else(|| {
+    AppError::Internal(anyhow::anyhow!(
+      "Failed to create database body from collab, db_collab_id: {}",
+      database_uuid_str,
+    ))
+  })?;
+  Ok((db_collab, db_body))
+}
+
+pub fn encode_collab_v1_bytes(
```

Comment:
what is the difference between collab_to_bin and encode_collab_v1_bytes ?

Reply:
`encode_collab_v1_bytes` spawns a tokio task blocking task to do it, while `encode_collab_v1_bytes` dont. I am removing this function, as it's no longer being used, (have switched to use `collab_to_bin` instead).